src/core.ts

```diff
@@ -4,37 +4,27 @@ import Joi from "joi"
 
 import { Issue } from "./types"
 
-export const syncIssue = async ({
-  issue,
-  graphql: gql,
-  project,
-  organization,
-}: {
-  issue: Issue
-  graphql: typeof OctokitGraphQL
-  project: { number: number; targetField: string; targetValue: string }
-  organization: string
-}) => {
-  /*
-
-    Step: Fetch the project's data so that we'll be able to create an item on it
-    for the issue given as input
-
-  */
-  const projectData = await gql<{
-    organization: {
-      projectNext: {
-        id: string
-        fields: {
-          nodes: {
-            id: string
-            name: string
-            settings?: string | null
-          }[]
-        }
+type ProjectData = {
+  organization: {
+    projectNext: {
+      id: string
+      fields: {
+        nodes: {
+          id: string
+          name: string
+          settings?: string | null
+        }[]
       }
     }
-  }>(
+  }
+}
+
+const fetchProjectData: (params: {
+  gql: typeof OctokitGraphQL
+  organization: string
+  number: number
+}) => Promise<ProjectData> = async ({ gql, organization, number }) => {
```

Comment:
Does this function need to be `async`? I think the `Promise<*>` alone would be enough since you're not `await`ing for anything

Reply:
We have a lint for this in other projects but sadly the linting configuration isn't shared between all repositories (have to copy-and-paste currently :-|)

Reply:
Good catch, thanks